packages/worker/src/index.ts

```diff
@@ -62,25 +61,34 @@ if (env.isProd()) {
 const server = http.createServer(app.callback())
 destroyable(server)
 
+let shuttingDown = false
 server.on("close", async () => {
-  if (env.isProd()) {
+  if (shuttingDown) {
+    return
+  }
+  shuttingDown = true
+  if (!env.isTest()) {
     console.log("Server Closed")
   }
   await redis.shutdown()
+  process.exit()
 })
 
+const shutdown = () => {
+  server.close()
+  server.destroy()
+}
+
 module.exports = server.listen(parseInt(env.PORT || 4002), async () => {
   console.log(`Worker running on ${JSON.stringify(server.address())}`)
   await redis.init()
 })
 
 process.on("uncaughtException", err => {
   console.error(err)
-  server.close()
-  server.destroy()
+  shutdown()
```

Comment:
Could we add a “bb-alert:” log line here, so we can send notifications when this happens

Reply:
It might be worth passing an exit code to the shutdown function, so it defaults to 0 but is set to 1 here